docs/source/reference/pysupported.rst

```diff
@@ -307,12 +307,72 @@ objects of different types, even if the types are compatible (for example,
    made to the set will not be visible to the Python interpreter until
    the function returns.
 
+dict
+----
+
+.. warning::
+  ``numba.typed.Dict`` is an experimental feature.  The API may change
+  in the future releases.
+
+Numba does not directly support the Python ``dict`` because it is an untyped container
+that can have any Python types as members.  To generate efficient machine code,
+Numba needs to know the types of the keys and the values, respectively.
+To achieve that, Numba implements the ``numba.typed.Dict``,
+a typed dictionary for which the user must explicitly declare the key-type
+and the value-type using ``Dict.empty()``. This typed dictionary is similar to
+the Python ``dict``.  It implements the ``collections.MutableMapping``
```

Comment:
"This typed dictionary is similar to" ==> "This typed dictionary has the same API as"